src/main/scala/sigmastate/eval/RuntimeCosting.scala

```diff
@@ -945,14 +945,7 @@ trait RuntimeCosting extends SigmaLibrary with DataCosting with Slicing { IR: Ev
           case _: BoxElem[_] => element[CostedBox].asElem[Costed[Any]]
           case _ => costedElement(eAny)
         }
-        val condC = node.condition match {
-          case Terms.Lambda(_, Seq((id, _)), _, Some(body)) =>
-            fun { x: Rep[Costed[Any]] => evalNode(ctx, env + (id -> x), body) }
-          case FuncValue(Seq((id, _)), body) =>
-            fun { x: Rep[Costed[Any]] => evalNode(ctx, env + (id -> x), body) }
-          case v =>
-            evalNode(ctx, env, v).asRep[CostedFunc[Unit, Any, SType#WrappedType]].func
-        }
+        val condC = evalNode(ctx, env, node.condition).asRep[CostedFunc[Unit, Any, SType#WrappedType]].func
```

Comment:
@greenhat I use method asRep(...) instead of .toRep because the later comes with unnecessary allocation of wrapper implicit class.

Reply:
@ergomorphic Fixed.